### PR TITLE
[BTC] Minor refactoring in fee handling.

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

Cleanup/rework of Fee computation.
- Some small change in behaviour:
  -- Minor change in max_amount amount computation
      https://github.com/trustwallet/wallet-core/pull/963/commits/95b3f6d80876d65f2b5f90dc5d8e397d1a73ef73
      https://github.com/trustwallet/wallet-core/pull/963/commits/df0a4a27279ef284093505e685fb1795e5b53d70
  -- Minor change in fee computation (special dust case)  https://github.com/trustwallet/wallet-core/pull/963/commits/736fa786b6cf22739015db215c54cf4ae167d485
- Code rearrange and cleanup (no change in behaviour)
 -- Rework FeeCalculator.
- Extra verifications in tests
- Extra tests

In preparation of SegWit fee computation.

## Testing instructions

Unit tests.

## Types of changes

* Refactoring (no change in behavior)
* Minor change in behavior

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

File: src/Bitcoin/FeeCalculator.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include <TrustWalletCore/TWCoinType.h>
+#include <functional>
```
[the remark this answers is not in the document]
Comment (catenocrypt) [reply src/Bitcoin/FeeCalculator.h]: done

**Comment by catenocrypt** [comment]
PR ready, summary for review:

Cleanup/rework of Fee computation.
- Some small change in behaviour:
  -- Minor change in max_amount amount computation
      https://github.com/trustwallet/wallet-core/pull/963/commits/95b3f6d80876d65f2b5f90dc5d8e397d1a73ef73
      https://github.com/trustwallet/wallet-core/pull/963/commits/df0a4a27279ef284093505e685fb1795e5b53d70
  -- Minor change in fee computation (special dust case)  https://github.com/trustwallet/wallet-core/pull/963/commits/736fa786b6cf22739015db215c54cf4ae167d485
- Code cleanup (no change in behaviour)
- Extra verifications in tests
- Extra tests

File: tests/Bitcoin/TxComparisonHelper.cpp [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    }
+    for (auto i = 0; i < selected.size() && i < expectedAmounts.size(); ++i) {
+        if (expectedAmounts[i] != selected[i].amount()) {
+            ss << "Wrong UTXOs amount, pos " << i << " amount " << selected[i].amount() << " expected " << expectedAmounts[i] << std::endl;
+        }
+    }
+    return ss.str();
+}
+
+std::string verifyPlan(const TransactionPlan& plan, const std::vector<int64_t>& utxoAmounts, int64_t outputAmount, int64_t fee) {
```
Comment (vikmeup) [inline tests/Bitcoin/TxComparisonHelper.cpp:72]: is there a better way to handle errors? it strange to return `string`

Comment (catenocrypt) [reply tests/Bitcoin/TxComparisonHelper.cpp:72]: Well, gtest ASSERTs are valid only within testcases (TEST() {...}), not outside.  The test context could be passed to helper function, but that would require tricks to dig out the test context hidden in the TEST macro -- ugly (I hate macros!).
Another option is to simple print differences on stderr, and only return bool for assert.  I found it more informative to return the error message as string.  But this is test only, of course.
Any better suggestions, anyone?

Comment (vikmeup) [reply tests/Bitcoin/TxComparisonHelper.cpp:72]: Return `bool` sounds more understanding to me, and use logging for printing an error.

Comment (catenocrypt) [reply tests/Bitcoin/TxComparisonHelper.cpp:72]: It is a bit cleaner, right

**Pull request merged**